### ISSUE TITLE
Envenom

### ISSUE THREAD

**Issue by kaytotes** [Project-Epoch/bugtracker#2891]
**Spell ID's:** 32645

> Bad.
It does less damage than eviscerate with armor debuffs. 
Does not get the guaranteed crit from Cold Blood.
Consuming poison stacks without compensation makes it only usable as an execute but honestly its not worth the keybind slot.

> Doesn't benefit from the Vile Poisons Assassination Talent

> Doesn't proc the Relentless Strikes Assassination talent.

> cold blood works with eviscerate but not with envenom

**Comment by zana244** [comment]
With its new implementation (DP consumed doses gives % proc buff), you get the buff even if your envenom misses/gets parried/dodged/blocked.
Vile poisons seems to work. 5 DP Envenom without talent hits for 820, talented it does 1176 which is close to 820 * 1.2 * 1.2 = 1180
Still does not proc relentless strikes, and cold blood does nothing for envenom.

**Comment by Hatson22** [comment]
Update:
1) Current version of Envenom on internal does in fact proc Relentless Strikes.
2) Current version of Envenom on internal does in fact work with Cold Blood.
3) Current version of Envenom on internal does work with Vile Poisons.

The only "bug" left to consider is if you want to keep the +10% poison proc buff on miss/parry/dodge

**Comment by Shadownflame** [comment]
This spell broke with the last update, does half the damage it used to. In a buffed environment it actually does less dmg now than the 4 ticks of 5/5 deadly poison would, forget any DPS gain from pressing it. I can't pinpoint what happened, there were no announced rogue changes, but it seems it just stopped scaling with effects like the stormstrike debuff or CoE. This renders the ability completely useless.
Note, in the previous state it properly scaled up with everything except Find Weakness.

**Comment by ulmetrs** [comment]
> This spell broke with the last update, does half the damage it used to. In a buffed environment it actually does less dmg now than the 4 ticks of 5/5 deadly poison would, forget any DPS gain from pressing it. I can't pinpoint what happened, there were no announced rogue changes, but it seems it just stopped scaling with effects like the stormstrike debuff or CoE. This renders the ability completely useless. Note, in the previous state it properly scaled up with everything except Find Weakness.

There are some core changes that could possibly effect this but can be difficult to pinpoint without very specific details of what is not behaving as expected.  Is the base damage with no buffs working as expected, does it get the correct bonuses from certain mods but not others, etc.  Having tested the core changes on stock wrath this does scale with damage taken effects such as elemental shield, but would need more info to continue the search.

**Comment by Shadownflame** [comment]
The base damage appears fine. In an unbuffed environment envenom damage matches the 4 ticks (full duration) of 5 stack DP at 5 combo points, multiplied by the 20% from talent. 
It a buffed environment, however, it doesn't seem to benefit from any of the debuffs which are meant to buff it (Coe, stormstrike, misery), and doesn't even match the base formula of 4ticks of DP +20%, it deals less than that. Eredun confirmed it's not reading the DP damage properly somehow and he is in progress of resolving that.

Damage buffs such as the attack power component of DP or buff% mods like Ferocious inspiration yet to be tested.

**Comment by ulmetrs** [comment]
Cool will dig into this when I have some time later today

**Comment by ulmetrs** [comment]
Doing some basic tests with the wrath stock client and worn daggers and deadly poison IX, envenom ability does:
1278 dmg ~(215 base points * 5 + 0.45 * AP).
Adding curse of the elements for 13% dmg moves this to 1444 damage as expected.
Adding the 20% talents and the numbers change as expected:
1533 dmg -> 1732 dmg with curse.

Other spells like elemental shield also multiply as expected.

A few things:
-Envenom does not do as much immediate damage as deadly poison 9 over its entire duration, the formula is independent and is based off base points and attack power (at least from the dbc and core)
-I did not do this on epoch server so I can't 100% verify this works as expected, though the epoch spell dbc for envenom is identical to stock wrath.
-I cannot test other auras mentioned using stock wrath as they are different: Stormstrike debuff only effects shaman damage in wrath, misery only effects priest spells spell power benefit etc.

**Comment by ulmetrs** [comment]
Seems envenom damage is overridden by a script hook so dbc info is irrelevant, awaiting more info

**Comment by Shadownflame** [comment]
> Seems envenom damage is overridden by a script hook so dbc info is irrelevant, awaiting more info

Yes epoch reverted every wotlk ability to tbc or custom tbc/wrath state, in case of envenom it's the latter. 
This is epoch version of envenom 

<img width="737" alt="Image" src="https://github.com/user-attachments/assets/69da02cd-8805-4942-8a72-297cd92a0939" />

On original tbc, envenom had two ranks, rank one at lvl 62 matched with the damage of deadly poison rank V (lvl 60) as a base, while rank 2 at 70 matched with the dmg of deadly poison rank VII as a base (as shown on wowhead tbc). Both had 15% AP scaling instead of the wrath 45% at 5 stacks.

The epoch version uses DP damage as a base much like tbc, except it calculates with whatever deadly poison you have applied instead of flat values so it has no ranks (which is handy because I think deadly poison rank V is a book we get later here).
As far as I know the 15% AP scaling at 5 stacks was changed (need Eredun to confirm), experimentally it's near 9-10% but the exact value wasn't posted on the website.
So envenom here deals (5 times DP base damage + whatever AP) before any modifiers at 5 cbp and 5 stacks, which checks out to be the equivalent of 4 ticks of 5 stack DP recorded in game.

On top of this, epoch retained a modified version of the 3.3.5 increased poison application chance after Envenom is used, it's a flat 10% extra chance here regardless of combo points, which is probably why the whole thing is scripted, but that shouldn't affect the damage calculations in any way.

For reference numbers, Deadly poison IV (lvl54) which is currently the highest available strength, deals 108 base damage over 12 seconds, or ~129 with Vile poisons, per stack. At five stacks this is expected to be 540 base damage without Vile poisons or ~645 damage with it, plus AP scaling.
With a test character at 784 AP it produces deadly poison ticks of 155 at 5 stacks (31 per stack) without Vile poisons which maths out to be 620 total damage or 80 dmg AP contribution (80/784= 10.2%)
With Vile poisons it goes up to 185 per tick (37 per stack), for a total of 740 damage which is a bonus of 95 dmg from AP (95/784=12.1%)
Single digit rounding errors aside (private servers eat decimals), this means Vile poisons buffs DP as intended.

Envenom with 5 stacks of this deadly poison (620 total damage/12sec) with no Vile poisons deals precisely 620 damage in an unbuffed environment.
Vile poisons properly buffs this to 740x1.2= 888 envenom damage (740 total dmg/12sec from dp)

Scenario 1) Apply stormstrike, misery and Coe, deadly poison 5 stack damage goes up to 230/tick or 920 dmg /12 seconds, as expected (1.13x1.05x1.05=1.2458 modifier times 740). Envenom damage still 888 instead of the expected 920x1.2=1104 before envenom's own modifiers and 1104x1.2458=1375 after envenom takes the dip from these buffs too. So it is not only not receiving benefit from the debuffs itself, it's also not considering the buffs deadly poison received from these effects.

Scenario 2) Apply sanctity aura (improved) and ferocious inspiration for a total of +5% damage (this shows up on your char sheet as a 1.05 modifier).
The 740 total damage went up to 780 or 195 per tick. Envenom dmg increased to 936, which is 780x1.2. This means Envenom considers the extra 5% dmg DP received from these buffs, but isn't receiving the 5% extra damage on its own behalf as it should. Expected damage should be 936x1.05= 982.

Testing concludes:

- Envenom is missing a roughly 55% extra damage from the double dipping 24.5% raid debuffs as it's not considering their dmg boost to DP nor it is receiving any benefit itself (should receive both dips)
- Envenom is missing a further 5% damage from raid buffs on its own behalf
- Envenom is still missing the 10% damage mod from Find Weakness proc (known issue from earlier)
- Deadly poison AP scaling needs confirmation from Devs

**Comment by ulmetrs** [comment]
The problem may be identified and systemic, working out the best course of action now.  The short of it is the script hook used is after the damage bonus calcs when we need one before.

**Comment by ghost** [comment]
Next deploy

And yes I can confirm the 2% AP scaling for Deadly Poison (per stack, hence the 10% estimate).
